src/datadog/glob.cpp

```diff
@@ -18,8 +19,11 @@ bool glob_match(StringView pattern, StringView subject) {
   Index next_p = 0;  // next [p]attern index
   Index next_s = 0;  // next [s]ubject index
 
-  while (p < pattern.size() || s < subject.size()) {
-    if (p < pattern.size()) {
+  const size_t p_size = pattern.size();
```

Comment:
nit: I can't comment on the relevant line, but instead of `using Index = std::size_t;` it would make more sense to use `using Index = StringView::size_type;`, it should work as expected with both `absl::string_view` and `std::string_view`. In practice it doesn't make a difference.